From 8767c1b2ad391d2216fec56ce23a6dc56a4079ff Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Fri, 6 Aug 2021 12:32:01 +0300
Subject: [PATCH 46/49] gpio: s32gen1: Set IRQ chip name based on device name

commit 8767c1b2ad391d2216fec56ce23a6dc56a4079ff from
https://source.codeaurora.org/external/autobsps32/linux

EIRQs were unmasked correctly because the same instance of
struct irq_chip was used for both GPIO instances.

Upstream-Status: Pending

Issue: ALB-7508, ALB-7507
Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/gpio/gpio-siul2-s32gen1.c | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/drivers/gpio/gpio-siul2-s32gen1.c b/drivers/gpio/gpio-siul2-s32gen1.c
index b36f9b0474eb..a0f31ae53498 100644
--- a/drivers/gpio/gpio-siul2-s32gen1.c
+++ b/drivers/gpio/gpio-siul2-s32gen1.c
@@ -124,6 +124,7 @@ struct siul2_gpio_dev {
 	struct regmap *ipadmap;
 	struct regmap *irqmap;
 	struct gpio_chip gc;
+	struct irq_chip irq;
 	spinlock_t lock;
 };
 
@@ -459,14 +460,6 @@ static void siul2_gpio_irq_mask(struct irq_data *data)
 	spin_unlock_irqrestore(&gpio_dev->lock, flags);
 }
 
-static struct irq_chip siul2_gpio_irq_chip = {
-	.name			= "siul2-gpio",
-	.irq_ack		= siul2_gpio_irq_mask,
-	.irq_mask		= siul2_gpio_irq_mask,
-	.irq_unmask		= siul2_gpio_irq_unmask,
-	.irq_set_type		= siul2_gpio_irq_set_type,
-};
-
 static const struct regmap_config siul2_regmap_conf = {
 	.val_bits = 32,
 	.reg_bits = 32,
@@ -895,6 +888,13 @@ int siul2_gpio_probe(struct platform_device *pdev)
 		sizeof(*gpio_dev->pin_dir_bitmap);
 	gpio_dev->pin_dir_bitmap = devm_kzalloc(&pdev->dev, bitmap_size,
 						GFP_KERNEL);
+	gpio_dev->irq = (struct irq_chip) {
+		.name			= dev_name(&pdev->dev),
+		.irq_ack		= siul2_gpio_irq_mask,
+		.irq_mask		= siul2_gpio_irq_mask,
+		.irq_unmask		= siul2_gpio_irq_unmask,
+		.irq_set_type		= siul2_gpio_irq_set_type,
+	};
 
 	gc->parent = &pdev->dev;
 	gc->label = dev_name(&pdev->dev);
@@ -908,7 +908,7 @@ int siul2_gpio_probe(struct platform_device *pdev)
 	gc->owner = THIS_MODULE;
 
 	girq = &gc->irq;
-	girq->chip = &siul2_gpio_irq_chip;
+	girq->chip = &gpio_dev->irq;
 	girq->parent_handler = NULL;
 	girq->num_parents = 0;
 	girq->parents = NULL;
-- 
2.17.1

